source/ngf-mtl/impl.cpp

```diff
@@ -2688,3 +2725,17 @@ uintptr_t ngf_get_mtl_sampler_handle(ngf_sampler sampler) NGF_NOEXCEPT {
 uint32_t ngf_get_mtl_pixel_format_index(ngf_image_format format) NGF_NOEXCEPT {
   return (uint32_t)get_mtl_pixel_format(format).format;
 }
+
+uintptr_t ngf_get_mtl_device() NGF_NOEXCEPT {
+  return (uintptr_t)(void*)MTL_DEVICE;
+}
+
+void ngf_mtl_set_sample_attachment_for_next_render_pass( ngf_cmd_buffer cmd_buffer, uintptr_t sample_buf_attachment_descriptor ) NGF_NOEXCEPT
+{
+  cmd_buffer->sample_buf_attachment_for_next_render_pass = static_cast< MTL::RenderPassSampleBufferAttachmentDescriptor* >( (void*)sample_buf_attachment_descriptor );
```

Comment:
probably want to use ngf_id::add_retain here (and below) to make sure your descriptor doesnt get yoinked

Reply:
Good idea. To mitigate it, I mention that the `sample_buf_attachment_descriptor` should already be retained when passing in the `nicegraf-mtl-handles.h`, in the comments. I'll change it.